projects/auth0-angular/src/lib/auth.interceptor.spec.ts

```diff
@@ -107,6 +111,21 @@ describe('The Auth HTTP Interceptor', () => {
         'Bearer access-token'
       );
     }));
+
+    it('attach the access token when the configuration uri is a string with a wildcard', fakeAsync((
+      done
+    ) => {
+      httpClient.get('/basic-api/startsWith?hello=world').subscribe(done);
```

Comment:
IMO, the use of the wildcard does not have any impact on URLs with or without a query string. The query string should be ignored, not considered when trying to match the endpoint. The path is the only portion we should use to match. Maybe the tests are a bit confusing because all of them include a query string value.

We could improve that to give the right message. A few examples:

- With a rule of `'/api/photos'` it will match only exact paths, like `/api/photos` and `/api/photos/` (is there any difference between both?? I don't think so).
- With a rule of `'/api/photos*'` it will match anything that starts with that, like `/api/photos`, `/api/photos/`, `/api/photos/31`, `/api/photos/me/summer/2020`.

Reply:
I updated the tests here with more representative URLs, and refactored out the assertion to a common function.

I updated the regex ones too, even though they're about to be removed in another PR.